source/ui/datasources/ui_blockedplayers_datasource.cpp

```diff
@@ -0,0 +1,46 @@
+#include "ui_precompiled.h"
+#include "kernel/ui_common.h"
+#include "datasources/ui_blockedplayers_datasource.h"
+#include "kernel/ui_syscalls.h"
+
+#define TABLE_NAME "blockedplayers"
+
+static int blocklistsize() {
+	int num = 0;
+	for (int i = 0; i < MAX_BLOCKS; i++) {
+		if ((*WSWUI::UI_IMPORT.GetBlocklist())[i].steamid != 0) {
+			num++;
+		}
+	}
+	return num;
+}
+namespace WSWUI
+{
+    BlockedPlayersDataSource::BlockedPlayersDataSource():Rocket::Controls::DataSource( "blockedplayers" )
+	{
+
+	}
+	
+	void BlockedPlayersDataSource::GetRow( Rocket::Core::StringList &row, const Rocket::Core::String&, int row_index, const Rocket::Core::StringList& cols )
+	{
+		if( row_index < 0 || (size_t)row_index > blocklistsize() )
```

Comment:
For each row, you are calculating the blocklist size again.
Possible twice even, depending on how librocket iterates over rows.